src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -3500,14 +3500,14 @@ internal Collection<PSObject> NewItem(string[] paths, string name, string type,
                     {
                         if (content == null)
                         {
-                            throw PSTraceSource.NewArgumentNullException(SessionStateStrings.NewItemValueNotSpecified, path);
+                            throw PSTraceSource.NewArgumentNullException(nameof(content), SessionStateStrings.NewItemValueNotSpecified, path);
                         }
 
                         string targetPath = content.ToString();
 
                         if (string.IsNullOrEmpty(targetPath))
                         {
-                            throw PSTraceSource.NewArgumentNullException(SessionStateStrings.PathNotFound, targetPath);
+                            throw PSTraceSource.NewArgumentNullException(nameof(targetPath), SessionStateStrings.PathNotFound, targetPath);
```

Comment:
I think it was and is not correct.
```suggestion
                            throw PSTraceSource.NewArgumentNullException(nameof(targetPath), SessionStateStrings.NewItemValueNotSpecified, path);
```

Reply:
I think it is dead code.

Reply:
Really the New-Item parameter is `Value` and type of the parameter is `object`. So user can set an object here. I agree that it is an edge case but for consistency we should fix and test the code path. (You could thing about that the Value could get more complicated expression in future, maybe scriptblock?)